client/client.go

```diff
@@ -334,7 +328,11 @@ func main() {
 		}
 	}
 
-	signedResp := resp.GetSignedResponse()
+	var signedResp bpb.BootstrapDataSigned
+	if err := proto.Unmarshal(resp.GetSerializedSignedResponse(), &signedResp); err != nil {
```

Comment:
Not 100% sure if the unmarshal here would always work if the client and the server have a slightly different  Protobuf versions? Maybe include Charles as reviewer too? another option might be gRPC might have some libraries to handle this situation? other than this, LGTM.

Reply:
Protos should always be backwards compatible. (unless changed incorrectly). I'd advise we err on the side of keeping our protos backwards compatible. If the client and server protos aren't compatible we'd have pretty big problems...

Reply:
+1 to maintaining the protos in a backwards compatible fashion. [The protobuf best practices](https://protobuf.dev/programming-guides/dos-donts/) provide a set of guidelines for doing just that.

In fact, this PR is fixing a best practice violation, specifically [reliance on stable serialization, which is something you should never do](https://protobuf.dev/programming-guides/dos-donts/#serialization-stability).

The current implementation relies on unmarshaling (previously performed on line 126) to produce exactly the same bytes that were used when the signature was created. This is inherently unreliable since the signature is created by a different binary (i.e. the server) and at a different time (i.e. it may have had a different message definition).

Basically, the current implementation cannot reliably verify valid signatures, which this new approach fixes.